irohad/model/converters/impl/pb_query_factory.cpp

```diff
@@ -169,6 +168,34 @@ namespace iroha {
         return val;
       }
 
+      protocol::Pager PbQueryFactory::serializePager(
+          const model::Pager &pager) const {
+        protocol::Pager pb_pager;
+        // Empty tx_hash is allowed in Pager
+        if (std::all_of(pager.tx_hash.begin(), pager.tx_hash.end(), [](auto x) {
+              return x == 0;
+            })) {
+          pb_pager.set_tx_hash("");
+        } else {
+          pb_pager.set_tx_hash(pager.tx_hash.to_string());
+        }
+        pb_pager.set_limit(pager.limit);
+        return pb_pager;
+      }
+
+      Pager PbQueryFactory::deserializePager(
+          const protocol::Pager &pb_pager) const {
+        Pager pager;
+        if (auto tx_hash =
+                stringToBlob<hash256_t::size()>(pb_pager.tx_hash())) {
+          pager.tx_hash = *tx_hash;
+        } else if (not pb_pager.tx_hash().empty()) {
+          log_->warn("pager transaction hash broken");
```

Comment:
Yes, so the condition is `NOT pb_pager.tx_hash().empty()`
And at first, I expected that regarding invalid length hash as empty, as not specification behavior. But finally, I've found it's difficult because it's no way to remember the length of hash when converting from invalid length protobuf string hash to strict typed `hash256_t`.

Reply:
If not knowing the length of protobuf size, the query hash can be changed and signature will be broken.